editor/editor_vcs_interface.cpp

```diff
@@ -152,6 +260,191 @@ String EditorVCSInterface::get_vcs_name() {
+	DiffFile df;
+	df.new_file = p_diff_file["new_file"];
+	df.old_file = p_diff_file["old_file"];
+	df.diff_hunks = List<DiffHunk>();
+	Array diff_hunks = p_diff_file["diff_hunks"];
+	for (int i = 0; i < diff_hunks.size(); i++) {
+		DiffHunk dh = _convert_diff_hunk(diff_hunks[i]);
+		df.diff_hunks.push_back(dh);
+	}
+	return df;
+}
+
+EditorVCSInterface::Commit EditorVCSInterface::_convert_commit(Dictionary p_commit) {
+	EditorVCSInterface::Commit c;
+	c.msg = p_commit["message"];
+	c.author = p_commit["author"];
+	c.time = p_commit["when"]; // Epoch time in seconds
+	c.hex_id = p_commit["id"];
+	return c;
+}
+
+EditorVCSInterface::StatusFile EditorVCSInterface::_convert_status_file(Dictionary p_status_file) {
+	StatusFile sf;
+	sf.file_path = p_status_file["file_path"];
+	sf.change_type = (ChangeType)(int)p_status_file["chanage_type"];
+	sf.area = (TreeArea)(int)p_status_file["area"];
+	return sf;
+}
+
+void EditorVCSInterface::_not_implemented_function(String p_function) {
+	ERR_PRINT("Selected VCS plugin does not implement " + p_function + " function");
+}
+
+void EditorVCSInterface::_bind_methods() {
+	// Proxy end points that act as fallbacks to unavailability of a function in the VCS addon
+	ClassDB::bind_method(D_METHOD("_initialize", "project_root_path"), &EditorVCSInterface::_initialize);
+	ClassDB::bind_method(D_METHOD("_is_vcs_initialized"), &EditorVCSInterface::_is_vcs_initialized);
+	ClassDB::bind_method(D_METHOD("_get_vcs_name"), &EditorVCSInterface::_get_vcs_name);
+	ClassDB::bind_method(D_METHOD("_shut_down"), &EditorVCSInterface::_shut_down);
+	ClassDB::bind_method(D_METHOD("_get_project_name"), &EditorVCSInterface::_get_project_name);
+	ClassDB::bind_method(D_METHOD("_get_modified_files_data"), &EditorVCSInterface::_get_modified_files_data);
+	ClassDB::bind_method(D_METHOD("_commit", "msg"), &EditorVCSInterface::_commit);
+	ClassDB::bind_method(D_METHOD("_get_file_diff", "file_path", "area"), &EditorVCSInterface::_get_file_diff);
+	ClassDB::bind_method(D_METHOD("_stage_file", "file_path"), &EditorVCSInterface::_stage_file);
+	ClassDB::bind_method(D_METHOD("_unstage_file", "file_path"), &EditorVCSInterface::_unstage_file);
+	ClassDB::bind_method(D_METHOD("_discard_file", "file_path"), &EditorVCSInterface::_discard_file);
+	ClassDB::bind_method(D_METHOD("_get_previous_commits"), &EditorVCSInterface::_get_previous_commits);
+	ClassDB::bind_method(D_METHOD("_get_branch_list"), &EditorVCSInterface::_get_branch_list);
+	ClassDB::bind_method(D_METHOD("_checkout_branch", "branch"), &EditorVCSInterface::_checkout_branch);
+	ClassDB::bind_method(D_METHOD("_push"), &EditorVCSInterface::_push);
+	ClassDB::bind_method(D_METHOD("_pull"), &EditorVCSInterface::_pull);
+	ClassDB::bind_method(D_METHOD("_fetch"), &EditorVCSInterface::_fetch);
+	ClassDB::bind_method(D_METHOD("_set_up_credentials"), &EditorVCSInterface::_set_up_credentials);
+	ClassDB::bind_method(D_METHOD("_get_line_diff", "file_path", "text"), &EditorVCSInterface::_get_line_diff);
```

Comment:
Shouldn't these be bound as virtual methods so that they appear in the docs and can be overridden from script?